src/dbus_stats.rs

```diff
@@ -0,0 +1,342 @@
+            };
+
+            let pair = CurMaxPair{
+                cur: match &field_tuple[1] {
+                    Value::U32(n) => *n,
+                    _ => continue,
+                },
+                max: match &field_tuple[2] {
+                    Value::U32(n) => *n,
+                    _ => continue,
+                },
+            };
+
+            match name.as_str() {
+                "Bytes" => user.bytes = Some(pair),
+                "Fds" => user.fds = Some(pair),
+                "Matches" => user.matches = Some(pair),
+                "Objects" => user.objects = Some(pair),
+                _ => {} // ignore other fields
+            }
+        }
+    }
+
+    return Some(user);
+}
+                
+fn parse_user_accounting(owned_value: &OwnedValue) -> Option<HashMap<u32, DBusBrokerUserAccounting>> {
+    let value: &Value = owned_value;
+    let users_value = match value {
+        Value::Array(users_value) => users_value,
+        _ => return None,
+    };
+
+    let mut result = HashMap::new();
+    for user_value in users_value.iter() {
+        if let Some(user) = parse_user_struct(user_value) {
+            result.insert(user.uid, user);
+        }
+    }
+
+    Some(result)
+}
+
+/// Pull all units from dbus and count how system is setup and behaving
+pub async fn parse_dbus_stats(
+    connection: &zbus::Connection,
+) -> Result<DBusStats, Box<dyn std::error::Error + Send + Sync>> {
+    let proxy = StatsProxy::new(connection).await?;
+    let stats = proxy.get_stats().await?;
+
+    let dbus_stats = DBusStats{
+        serial: stats.serial(),
+        active_connections: stats.active_connections(),
+        incomplete_connections: stats.incomplete_connections(),
+        bus_names: stats.bus_names(),
+        peak_bus_names: stats.peak_bus_names(),
+        peak_bus_names_per_connection: stats.peak_bus_names_per_connection(),
+        match_rules: stats.match_rules(),
+        peak_match_rules: stats.peak_match_rules(),
+        peak_match_rules_per_connection: stats.peak_match_rules_per_connection(),
+
+        // attempt to parse dbus-broker specific stats
+        dbus_broker_peer_accounting: stats.rest().get("org.bus1.DBus.Debug.Stats.PeerAccounting")
+            .map(parse_peer_accounting)
+            .unwrap_or_default(),
+        dbus_broker_user_accounting: stats.rest().get("org.bus1.DBus.Debug.Stats.UserAccounting")
+            .map(parse_user_accounting)
+            .unwrap_or_default(),
+    };
+
+    Ok(dbus_stats)
+}
+
+/// Async wrapper than can update dbus stats when passed a locked struct
+pub async fn update_dbus_stats(
+    connection: zbus::Connection,
+    locked_machine_stats: Arc<RwLock<MachineStats>>,
+) -> anyhow::Result<()> {
+    let mut machine_stats = locked_machine_stats.write().await;
+    match parse_dbus_stats(&connection).await {
+        Ok(dbus_stats) => machine_stats.dbus_stats = Some(dbus_stats),
+        Err(err) => error!("dbus stats failed: {:?}", err),
+    }
```

Comment:
Lets get the machine_stats in the Ok(dbus_stats) clause
```suggestion
    match parse_dbus_stats(&connection).await {
        Ok(dbus_stats) => {
            let mut machine_stats = locked_machine_stats.write().await;
            machine_stats.dbus_stats = Some(dbus_stats)
        },
        Err(err) => error!("dbus stats failed: {:?}", err),
    }
```
This will ensure we hold the write lock for the shortest time possible and other concurrent collections can write asap